### ISSUE TITLE
Prediction

### ISSUE THREAD

**Issue by bernardogmorais** [Darel13712/ease_rec#1]
When I try to predict, the expected input is, according to my understanding:
- _train_: a data frame containing the train_df (df used to fit)
- _users_: list of users on the val/test set
- _products_: list of products on the val/test set that should be recommended to users _users_
- _k_: number of top k recommendations

But my predict returns a data frame containing K duplicate entries, with the same score _s_, and product _p_, for the user _u_

**Comment by Darel13712** [comment]
Your understanding seems to be correct.

I am not able to reproduce this on MovieLens, could you please provide an example so that I could reproduce this issue?

```python
from rs_datasets import MovieLens
ml = MovieLens()

model = EASE()
model.fit(ml.ratings)
model.predict(ml.ratings, [1], ml.ratings.item_id.unique(), 10)
```

```
   user_id  item_id     score
5        1      588  0.006195
9        1      318  0.005041
3        1     1200  0.004791
7        1     1387  0.004625
6        1     2762  0.004360
2        1     1036  0.004167
4        1     1610  0.003945
8        1     1968  0.003269
1        1      342  0.003246
0        1     1022  0.003181
```

Chances are, the list you provide for `items` argument contains duplicates. You are not supposed to provide a cartesian product of length `num_users * num_items` to get predictions for all users and all items. Instead, `users` argument defines which users get predictions and `items` argument defines the set of items to get recommendations from for **all** users. If you want to predict different items for different users, you have to rewrite it a bit.
